### PR TITLE
[Sol -> Yul] Implement if statements

### LINKED ISSUE #6709: [Sol -> Yul] Implement if statement

### PR DESCRIPTION
**Pull request by ekpyron**
Closes  #6709.

### CODE REVIEW

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp [hunk 1]
```diff
@@ -152,6 +152,27 @@ bool IRGeneratorForStatements::visit(TupleExpression const& _tuple)
 	return false;
 }
 
+bool IRGeneratorForStatements::visit(IfStatement const& _ifStatement)
```
Comment (leonardoalt) [inline libsolidity/codegen/ir/IRGeneratorForStatements.cpp:155]: Why not `endVisit`?

Comment (leonardoalt) [reply libsolidity/codegen/ir/IRGeneratorForStatements.cpp:155]: Ok nevermind

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp [hunk 2]
```diff
@@ -152,6 +152,27 @@ bool IRGeneratorForStatements::visit(TupleExpression const& _tuple)
...
+	_ifStatement.trueStatement().accept(*this);
+	m_code << "}\n";
+
+	if (_ifStatement.falseStatement())
+	{
+		m_code << "case 0 {\n";
+		_ifStatement.falseStatement()->accept(*this);
+		m_code << "}\n";
+	}
+	return false;
```
Comment (leonardoalt) [inline libsolidity/codegen/ir/IRGeneratorForStatements.cpp:171]: Do we want to be paranoid and revert on `default`?

Comment (ekpyron) [reply libsolidity/codegen/ir/IRGeneratorForStatements.cpp:171]: Hm... that probably can't be optimized away later, can it? We could use ``case 0`` and ``default`` instead... I'm not sure what's better. I was in fact also wondering whether it's better to use a ``switch`` or one ``if cond`` and one ``if iszero(cond)``, but I think it's easier to convert the switch to something else in the optimizer than the other way round...

Comment (chriseth) [reply libsolidity/codegen/ir/IRGeneratorForStatements.cpp:171]: I think `case 0` and `default` would be best.

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp [hunk 3]
```diff
@@ -152,6 +152,25 @@ bool IRGeneratorForStatements::visit(TupleExpression const& _tuple)
...
+bool IRGeneratorForStatements::visit(IfStatement const& _ifStatement)
+{
+	_ifStatement.condition().accept(*this);
+	string condition = expressionAsType(_ifStatement.condition(), *TypeProvider::boolean());
+
+	if (_ifStatement.falseStatement())
+	{
+		m_code << "switch " << condition << "\n" "case 0 {\n";
+		_ifStatement.falseStatement()->accept(*this);
+		m_code << "}\n" "default {\n";
```
Comment (ekpyron) [inline libsolidity/codegen/ir/IRGeneratorForStatements.cpp:164]: Do we have preferences on how to do this?
```
  m_code << "}\n";
  m_code << "default {\n";
```
and
```
  m_code << "}\n" << "default {\n";
```
are theoretically slower and
```
  m_code << "}\ndefault {\n";
```
looks just weird, so ``"}\n" "default {\n"`` (which is equivalent to the last one) seems best to me, but I'm happy to change.

**Comment by ekpyron** [comment]
Updated to ``case 0`` ``default``.

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp (hunk 3 above)
Comment (chriseth) [reply libsolidity/codegen/ir/IRGeneratorForStatements.cpp:164]: Readable beats speed and I would also not make any assumptions about speed for string literals...

**Review by chriseth — APPROVED** [review]

**Pull request merged**